cmd/zoekt-mirror-gitea/main.go

```diff
@@ -0,0 +1,315 @@
+		if hash[e] {
+			return true
+		}
+	}
+	return false
+}
+
+func filterRepositories(repos []*gitea.Repository, include []string, exclude []string, noArchived bool) (filteredRepos []*gitea.Repository) {
+	for _, repo := range repos {
+		if noArchived && *&repo.Archived {
+			continue
+		}
+		// FIXME: get topics from repos
+		// if (len(include) == 0 || hasIntersection(include, repo.Topics)) &&
+		// 	!hasIntersection(exclude, repo.Topics) {
+		// 	filteredRepos = append(filteredRepos, repo)
+		// }
+	}
+	return
+}
+
+func getOrgRepos(client *gitea.Client, org string, reposFilters reposFilters) ([]*gitea.Repository, error) {
+	var allRepos []*gitea.Repository
+	searchOptions := &gitea.SearchRepoOptions{}
+	// OwnerID
+	organization, _, err := client.GetOrg(org)
+	if err != nil {
+		return nil, err
+	}
+
+	searchOptions.OwnerID = organization.ID
+
+	for {
+		repos, resp, err := client.SearchRepos(*searchOptions)
+		if err != nil {
+			return nil, err
+		}
+		if len(repos) == 0 {
+			break
+		}
+
+		searchOptions.Page = resp.NextPage
+		repos = filterRepositories(repos, reposFilters.topics, reposFilters.excludeTopics, *reposFilters.noArchived)
+		allRepos = append(allRepos, repos...)
+		if resp.NextPage == 0 {
+			break
+		}
+	}
+	return allRepos, nil
+}
+
+func getUserRepos(client *gitea.Client, user string, reposFilters reposFilters) ([]*gitea.Repository, error) {
+	var allRepos []*gitea.Repository
+	searchOptions := &gitea.SearchRepoOptions{}
+	u, _, err := client.GetUserInfo(user)
+	if err != nil {
+		return nil, err
+	}
+	searchOptions.OwnerID = u.ID
+	for {
+		repos, resp, err := client.SearchRepos(*searchOptions)
+		if err != nil {
+			return nil, err
+		}
+		if len(repos) == 0 {
+			break
+		}
+
+		searchOptions.Page = resp.NextPage
+		repos = filterRepositories(repos, reposFilters.topics, reposFilters.excludeTopics, *reposFilters.noArchived)
+		allRepos = append(allRepos, repos...)
+		if resp.NextPage == 0 {
+			break
+		}
+	}
+	return allRepos, nil
+}
+
+func itoa(p int) string {
+	return strconv.Itoa(p)
+}
+
+func cloneRepos(destDir string, repos []*gitea.Repository) error {
+	for _, r := range repos {
+		host, err := url.Parse(r.HTMLURL)
+		if err != nil {
+			return err
+		}
+
+		config := map[string]string{
+			"zoekt.web-url-type": "gitea",
+			"zoekt.web-url":      r.HTMLURL,
+			"zoekt.name":         filepath.Join(host.Hostname(), r.FullName),
+
+			"zoekt.gitea-stars":       itoa(r.Stars),
+			"zoekt.gitea-watchers":    itoa(r.Watchers),
+			"zoekt.gitea-subscribers": itoa(r.Watchers), // FIXME: Get repo subscribers from API
+			"zoekt.gitea-forks":       itoa(r.Forks),
+
+			"zoekt.archived": marshalBool(r.Archived),
+			"zoekt.fork":     marshalBool(r.Fork),
+			"zoekt.public":   marshalBool(r.Private || r.Internal), // count internal repos as private
+		}
+		dest, err := gitindex.CloneRepo(destDir, r.FullName, r.CloneURL, config)
+		if err != nil {
+			return err
+		}
+		if dest != "" {
+			fmt.Println(dest)
```

Comment:
debug logs?

Reply:
This is copy/paste from the GitHub mirror code